src/main/scala/nl/knaw/dans/lib/string/package.scala

```diff
@@ -0,0 +1,51 @@
+/**
+ * Copyright (C) 2016 DANS - Data Archiving and Networked Services (info@dans.knaw.nl)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package nl.knaw.dans.lib
+
+package object string {
+
+  implicit class StringExtensions(val s: String) extends AnyVal {
+
+    /**
+     * Return true when all characters in the `String` are classified as ''whitespace'', false otherwise.
+     *
+     * @return whether all characters are whitespace
+     */
+    def isBlank: Boolean = {
+      s match {
+        case null | "" => true
+        case _ => s.forall(Character.isWhitespace)
```

Comment:
`s == null || s.trim == ""` would do the same job, I suppose more efficiently. There is some apache-commons or something like that that does the same. If commonly available drop this one.

Reply:
not sure yet. This function is indeed also implemented in Apache's `StringUtils.isBlank`, but we need this function here because it is used below in `toOption` and `emptyIfBlank`. We could switch to calling Apache, but that means having Apache Commons as a dependency on this project. I'd prefer to keep the size of this library as small as possible and with as few dependencies as possible. Since this isn't complex behavior (in contrast to the dependencies that were introduced to support `DebugEnhancedLogging`), I'd rather have this function here than include another library!

@jo-pol @DANS-KNAW/easy let me know what you think!